sqld/src/rpc/replication_log.rs

```diff
@@ -268,12 +268,18 @@ impl ReplicationLog for ReplicationLogService {
         {
             Ok(Ok(Some(snapshot))) => {
                 BLOCKING_RT.spawn_blocking(move || {
-                    let mut frames = snapshot.frames_iter_from(offset);
+                    let size_after = snapshot.header().size_after;
+                    let mut frames = snapshot.frames_iter_from(offset).peekable();
                     loop {
                         match frames.next() {
-                            Some(Ok(frame)) => {
+                            Some(Ok(mut frame)) => {
+                                // this is the last frame we're sending for this snapshot, set the
+                                // frame_no
```

Comment:
If we rely on the order here, let's also comment it in the code that generates frames (frame_iter_from?)